### PR TITLE
[aotd] Alias of intermediate unwrap TensorAlias

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #147638

Bug was reported by internal user.

AOTD classified outputs that are aliases of intermediates of the graph in different categories.

...
- output is alias of intermediate which base is already output
- output is alias of intermediate which base is not in output

If we look at the fn:
```
def fn(x):
    ix = x + 1
    a = ix.transpose(0, 1)
    return a.detach(), a
```

output 0: detach view of alias a, where a is already output
output 1: alias of intermediate ix, then additional output ix will be added internally

output 0 base is TensorAlias(a) in this case, but could be Tensor.
Adding runtime unwrapping solves this problem.

Alternatively we should track base of a.detach() all the way to ix, in that case the base will be always a Tensor, not TensorAlias.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

### CODE REVIEW

File: torch/_functorch/_aot_autograd/runtime_wrappers.py [hunk 1]
```diff
@@ -218,7 +218,9 @@ def __init__(self, info, runtime_metadata, trace_joint):
     def __call__(self, orig_inputs, fw_outs, out):
         aliased_base_tensor = fw_outs[self.base_idx]
         return gen_alias_from_base(
-            aliased_base_tensor,
+            aliased_base_tensor.alias
+            if isinstance(aliased_base_tensor, TensorAlias)
+            else aliased_base_tensor,
```
Comment (bdhirsh) [inline torch/_functorch/_aot_autograd/runtime_wrappers.py:223]: synced offline - one very nice outcome would be if we can **entirely avoid** doing the alias-of-intermediate logic in the repro above. we only actually need to do view replay if we have two graph outputs that are **differentiable** aliases of each other, which is not true in this repro (one is detached)

**Review by bdhirsh — APPROVED** [review]
sgtm
